notification-service/src/main/java/com/redhat/parodos/notification/controller/NotificationRecordController.java

```diff
@@ -59,53 +59,70 @@ public class NotificationRecordController {
+					@ApiResponse(responseCode = "401", description = "Unauthorized"),
+					@ApiResponse(responseCode = "403", description = "Forbidden") })
 	@GetMapping
-	@ResponseStatus(HttpStatus.OK)
-	public PagedModel<NotificationRecordResponseDTO> getNotifications(@PageableDefault(size = 100) Pageable pageable,
+	public ResponseEntity<Page<NotificationRecordResponseDTO>> getNotifications(
+			@PageableDefault(size = 100) Pageable pageable,
 			@RequestParam(value = "state", required = false) State state,
 			@RequestParam(value = "searchTerm", required = false) String searchTerm) {
-		if (securityUtil == null || securityUtil.getUsername() == null) {
-			throw new ResponseStatusException(HttpStatus.BAD_REQUEST, "No username associated with this request");
-		}
-		Page<NotificationRecord> notificationsRecordPage = this.notificationRecordService
-				.getNotificationRecords(pageable, securityUtil.getUsername(), state, searchTerm);
-		return this.notificationRecordPagedResourcesAssembler.toModel(notificationsRecordPage,
-				this.notificationAssemblerDTO);
+
+		Page<NotificationRecord> notificationsPage = this.notificationRecordService.getNotificationRecords(pageable,
+				securityUtil.getUsername(), state, searchTerm);
+		Page<NotificationRecordResponseDTO> notificationsDtoPage = notificationsPage
+				.map(NotificationRecord -> modelMapper.map(NotificationRecord, NotificationRecordResponseDTO.class));
+
+		return ResponseEntity.ok(notificationsDtoPage);
 	}
 
 	/**
 	 * Returns unread notifications for a user.
 	 */
+	@io.swagger.v3.oas.annotations.Operation(
+			summary = "Returns the amount of the unread notification records for the user")
+	@ApiResponses(value = {
+			@ApiResponse(responseCode = "200", description = "Successfully retrieved the amount of notifications"),
+			@ApiResponse(responseCode = "401", description = "Unauthorized"),
+			@ApiResponse(responseCode = "403", description = "Forbidden") })
 	@GetMapping("/count")
 	@ResponseStatus(HttpStatus.OK)
 	public int countUnreadNotifications(@RequestParam State state) {
 		return this.notificationRecordService.countNotificationRecords(securityUtil.getUsername(), state);
 	}
 
+	@ApiResponses(value = { @ApiResponse(responseCode = "200", description = "Succeeded"),
+			@ApiResponse(responseCode = "401", description = "Unauthorized"),
+			@ApiResponse(responseCode = "404", description = "Not found") })
+	@io.swagger.v3.oas.annotations.Operation(summary = "Updates the specified notification record with user operation")
 	@PutMapping("/{id}")
 	public NotificationRecordResponseDTO updateNotificationStatusById(@PathVariable("id") UUID id,
 			@RequestParam Operation operation) {
 		NotificationRecord notificationRecord = this.notificationRecordService.updateNotificationStatus(id, operation);
-		return notificationAssemblerDTO.toModel(notificationRecord);
+		return modelMapper.map(notificationRecord, NotificationRecordResponseDTO.class);
```

Comment:
TBH, I prefer to use the toModel() is much better for testing.

Reply:
Changed